website/views.py

```diff
@@ -289,33 +319,62 @@ def filter(request,  category=None, tutorial=None, minute_range=None, second_ran
+        questions = Question.objects\
+                            .filter(category=category)\
+                            .filter(tutorial=tutorial)\
+                            .filter(minute_range=minute_range)\
+                            .filter(second_range=second_range)\
+                            .order_by('date_created').reverse()
     elif tutorial is None:
-        questions = Question.objects.filter(category__name=category).order_by('date_created').reverse()
+        questions = Question.objects\
+                            .filter(category__name=category)\
+                            .order_by('date_created').reverse()
     elif minute_range is None:
-        questions = Question.objects.filter(category__name=category).filter(sub_category=tutorial).order_by('date_created').reverse()
+        questions = Question.objects\
+                            .filter(category__name=category)\
+                            .filter(sub_category=tutorial)\
+                            .order_by('date_created').reverse()
     else:
-        questions = Question.objects.filter(category=category).filter(sub_category=tutorial).filter(minute_range=minute_range).order_by('date_created').reverse()
+        questions = Question.objects\
+                            .filter(category=category)\
+                            .filter(sub_category=tutorial)\
+                            .filter(minute_range=minute_range)\
+                            .order_by('date_created').reverse()
 
     if 'qid' in request.GET:
-        context['qid']  = int(request.GET['qid'])
+        context['qid'] = int(request.GET['qid'])
      
     categories = FossCategory.objects.filter(name=category)
-    sub_category = SubFossCategory.objects.filter(name = tutorial)
+    sub_category = SubFossCategory.objects.filter(name=tutorial)
     try:
-        name = SubFossCategory.objects.get(name = tutorial)
+        name = SubFossCategory.objects.get(name=tutorial)
         dict_context['name'] = name
     except:
         pass        
     dict_context = {
-            'questions':questions,
+            'questions': questions,
             'categories': categories,
             'tutorial': sub_category,
            }
         
-    return render(request, 'website/templates/filter.html',  dict_context)
+    return render(request, 'website/templates/filter.html', dict_context)
+
+def tags(request, tag=None):
+    # filter the questions according to their tags
+    tag_s = str(tag)    # converting tag from unicode to string type
+    context={}
+
+    questions = Question.objects.filter(tags__name__in=[tag_s])
+
+    context = {'questions': questions,
+                'tag': tag_s
+               }
+    return render(request, 'website/templates/filter_tags.html', context)
 
-# post a new question on to forums, notification is sent to mailing list team@fossee.in
+
+
+# post a new question on to forums,
+# notification is sent to mailing list team@fossee.in
```

Comment:
This should be a docstrings. Please refer to Python docs and add this info as a docstring for this function